src/app/core/entity-list/entity-list/entity-list.component.ts

```diff
@@ -178,6 +182,7 @@ export class EntityListComponent<T extends Entity>
     if (!this.allEntities) {
       // if no entities are passed as input, by default load all entities of the type
       await this.loadEntities();
+      this.exportFile()
```

Comment:
to avoid having to call this in multiple places - and for following the separation of concerns principle - I would suggest to move your `exportFile` method to the `ExportDataDirective` - or maybe even better to the DownloadService.

We have extracted the entityConstructur from the data objects in another place also, using something like `entityConstructor = records[0]?.getConstructor();`, so you could also get to the schema there.